cmd/bosun/conf/system.go

```diff
@@ -540,6 +601,67 @@ func (sc *SystemConf) GetElasticContext() expr.ElasticHosts {
 	return parseESConfig(sc)
 }
 
+// GetAzureMonitorContext returns a Azure Monitor API context which
+// contains the information needs to query the azure API
+func (sc *SystemConf) GetAzureMonitorContext() expr.AzureMonitorClients {
+	allClients := make(expr.AzureMonitorClients)
+	for prefix, conf := range sc.AzureMonitorConf {
+		clients := expr.AzureMonitorClientCollection{}
+		if conf.Concurrency == 0 {
+			clients.Concurrency = 10
+		} else {
+			clients.Concurrency = conf.Concurrency
+		}
+		clients.MetricsClient = insights.NewMetricsClient(conf.SubscriptionId)
+		clients.MetricDefinitionsClient = insights.NewMetricDefinitionsClient(conf.SubscriptionId)
+		clients.ResourcesClient = resources.NewClient(conf.SubscriptionId)
+		if conf.DebugRequest {
+			clients.ResourcesClient.RequestInspector, clients.MetricsClient.RequestInspector, clients.MetricDefinitionsClient.RequestInspector = azureLogRequest(), azureLogRequest(), azureLogRequest()
+		}
+		if conf.DebugResponse {
+			clients.ResourcesClient.ResponseInspector, clients.MetricsClient.ResponseInspector, clients.MetricDefinitionsClient.ResponseInspector = azureLogResponse(), azureLogResponse(), azureLogResponse()
+		}
+		ccc := auth.NewClientCredentialsConfig(conf.ClientId, conf.ClientSecret, conf.TenantId)
+		at, err := ccc.Authorizer()
+		if err != nil {
+			// Should not hit this since we check for authorizer errors in Validation
+			// This is checked before because this method is not called until the an expression is called
+			slog.Fatal("Azure conf: ", err)
+		}
+		clients.MetricsClient.Authorizer, clients.MetricDefinitionsClient.Authorizer, clients.ResourcesClient.Authorizer = at, at, at
+		allClients[prefix] = clients
+	}
+	return allClients
+}
+
+func azureLogRequest() autorest.PrepareDecorator {
```

Comment:
Comment this function and the following, add comment to attribute source of readme from the go azure sdk

Reply:
I don't think they hurt, and since the documentation is marking the experimental for now they could be handy. I would probably be better if more datasources had this I think